### ISSUE TITLE
llvm-config needs to report -D_DEBUG as a cxxflag when built using Microsoft's standard library in debug mode.

### ISSUE THREAD

**Issue by dabrahams** [llvm/llvm-project#83697]
The ABI of Microsoft's standard library (e.g. object layout of iterators and collections) differs between standard CMake `Debug` and `Release` builds; `Debug` builds use a less-efficient representation that catches more erroneous usage at runtime.  However, `llvm-config` from a `Debug` build does not put `-D_DEBUG` in its `--cxxflags` output, and `llvm-config` is often used to generate pkg-config files, which transmit the usage requirements for LLVM to clients  (there is also no way to know from `llvm-config` whether the LLVM build used `-D_DEBUG`, although `--build-mode` comes close). When a  client uses such a .pc file to build against LLVM it results in link errors with newer MSVC libraries (and silent ODR violations with older ones).

In short: when built against the MSVC standard library, if LLVM was built with `_DEBUG` defined, `-D_DEBUG` should be among the `--cxxflags` output from the corresponding `llvm-config`.

**Comment by dabrahams** [comment]
Actually I think this goes a little further: `_DEBUG` is too coarse; you need to report a value for `_ITERATOR_DEBUG_LEVEL`, which can be `0`, `1`, or `2`.  Also missing: definitions like `-D_MT` and `-D_DLL` that control which runtime library is linked against.
